### PR TITLE
Added Campaign Support for all api calls

### PR DESCRIPTION
**Pull request by desusai7**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/am/transform.js [hunk 1]
```diff
@@ -274,7 +277,10 @@ function responseBuilderSimple(
       traits = getFieldValueFromMessage(message, "traits");
       set(rawPayload, "event_properties", message.properties);
       if (traits) {
-        set(rawPayload, "user_properties", traits);
+        rawPayload.user_properties = {
+          ...rawPayload.user_properties,
+          ...traits
```
Comment (prabrishac) [inline v0/destinations/am/transform.js:282]: There could be conflict between `context.traits.name` and `campaign.name`

Comment (prabrishac) [reply v0/destinations/am/transform.js:282]: I think we can prepend the keys with `utm_`

Comment (desusai7) [reply v0/destinations/am/transform.js:282]: Yeah definitely that would cause a conflict. So will append the campaign keys with utm_

Comment (desusai7) [reply v0/destinations/am/transform.js:289]: Added those changes now

File: v0/destinations/am/transform.js [hunk 2]
```diff
@@ -213,6 +213,21 @@ function responseBuilderSimple(
 
   // 2. get campaign info (only present for JS sdk and http calls)
   const campaign = get(message, "context.campaign") || {};
+  const oldKeys = Object.keys(campaign);
+  // appends utm_ prefix to all the keys of campaign object. For example the `name` key in campaign object will be changed to `utm_name`
+  oldKeys.forEach(oldKey => {
+    delete Object.assign(campaign, { [`utm_${oldKey}`]: campaign[oldKey] })[
```
Comment (prabrishac) [inline v0/destinations/am/transform.js:219]: I think its more readable if we break the copy and deletion of keys into 2 separate operations.
Also I think, `campaign.name` should be mapped to `utm_campaign` as while creating the campaign object in JS SDK, we parse the query string like this only - `utm_campaign` to `campaign.name`. Rest are fine.

**Review by prabrishac — APPROVED** [review]
Looks good to me 👍

**Review by arnab-p — APPROVED** [review]

**Pull request merged**